File: app/src/main/java/com/ignis/android_cleanarchitecture/presentation/view/activity/AboutActivity.java
package com.ignis.android_cleanarchitecture.presentation.view.activity;

import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import com.ignis.android_cleanarchitecture.R;
import com.ignis.android_cleanarchitecture.databinding.ActivityAboutBinding;
import com.ignis.android_cleanarchitecture.presentation.listener.activity.AboutActivityListener;
import com.ignis.android_cleanarchitecture.presentation.presenter.activity.AboutActivityViewModel;
import com.ignis.android_cleanarchitecture.presentation.view.fragment.AboutFragment;

/**
 * About Activity
 *
 * @author Kensuke
 */
public class AboutActivity extends AppCompatActivity implements AboutActivityListener {

    private AboutActivityViewModel viewModel;

    public static Intent newInstance(Context context) {
        Intent intent = new Intent(context, AboutActivity.class);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityAboutBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_about);
        viewModel = new AboutActivityViewModel(this, this);
        binding.setViewModel(viewModel);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);

        if (savedInstanceState == null) {
            replaceAboutFragment();
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                viewModel.onClickBackIcon();
                break;
            default:
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    private void replaceAboutFragment() {
        AboutFragment fragment = AboutFragment.newInstance();

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.container_fragment, fragment);
        fragmentTransaction.commit();
    }

    /* AboutActivity.AboutActivityListener */

    @Override
    public void onPerformFinish() {
        finish();
    }

}

File: presentation/src/main/java/com/kazakago/cleanarchitecture/presentation/view/activity/AboutActivity.kt
package com.kazakago.cleanarchitecture.presentation.view.activity

import android.arch.lifecycle.ViewModelProviders
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import com.kazakago.cleanarchitecture.presentation.R
import com.kazakago.cleanarchitecture.presentation.listener.activity.AboutActivityViewModelListener
import com.kazakago.cleanarchitecture.presentation.presenter.activity.AboutActivityViewModel
import com.kazakago.cleanarchitecture.presentation.view.fragment.AboutFragment
import kotlinx.android.synthetic.main.activity_about.*

class AboutActivity : AppCompatActivity(), AboutActivityViewModelListener {

    companion object {
        fun createIntent(context: Context): Intent = Intent(context, AboutActivity::class.java)
    }

    private lateinit var viewModel: AboutActivityViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)
        viewModel = ViewModelProviders.of(this, AboutActivityViewModel.Factory(application)).get(AboutActivityViewModel::class.java)
        viewModel.listener = this

        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        if (savedInstanceState == null) {
            replaceAboutFragment()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        viewModel.listener = null
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> viewModel.onClickBackIcon()
        }
        return super.onOptionsItemSelected(item)
    }

    private fun replaceAboutFragment() {
        val fragment = AboutFragment.createInstance()

        val fragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.fragmentContainer, fragment)
        fragmentTransaction.commit()
    }

    /* AboutActivityViewModelListener */

}
